### PR TITLE
Fix the module not being able to set the DM flag because Synapse gives frozen data in response to `get_global`.

### PR DESCRIPTION
**Pull request by reivilibre**
This has been tested locally, with the receiver's client offline. (Element Web will 'helpfully' update `m.direct` as long as it sees the invite and the join — it doesn't need to be the one that created the join to do so.)

### CODE REVIEW

**Review by anoadragon453 — CHANGES_REQUESTED** [review]
lightning review :zap:

File: synapse_auto_accept_invite/__init__.py [hunk 1]
```diff
@@ -117,26 +117,31 @@ async def _mark_room_as_direct_message(
         from the perspective of the user `user_id`.
         """
 
-        # This dict of User IDs to lists of Room IDs
-        dm_map: Dict[str, List[str]] = (
+        # This dict of User IDs to tuples of Room IDs
+        # (get_global will return a frozendict of tuples as it freezes the data,
+        # but we should accept either frozen or unfrozen variants.)
+        dm_map: Dict[str, Tuple[str, ...]] = dict(
```
[the remark this answers is not in the document]
Comment (reivilibre) [reply synapse_auto_accept_invite/__init__.py:123]: Aye, that's why I do the awkward `tuple(...) + (x,)` thing below to append. Maybe we should've just cloned all the data we pass out of the module API, it's a bit messy and error prone, but oh well :/

[the remark this answers is not in the document]
Comment (anoadragon453) [reply synapse_auto_accept_invite/__init__.py:126]: > Maybe we should've just cloned all the data we pass out of the module API

I can see arguments for doing both... At least it fails loudly :)

**Review by anoadragon453 — APPROVED** [review]
lgtm!

**Pull request merged**